src/Periphery/LiFiDEXAggregator.sol

```diff
@@ -854,6 +861,70 @@ contract LiFiDEXAggregator is WithdrawablePeriphery {
         ISyncSwapPool(pool).swap(data, from, address(0), new bytes(0));
     }
 
+    /// @notice Performs a swap through KatanaV3 pools
+    /// @dev This function handles swaps through KatanaV3 pools.
+    /// @param stream [pool, direction, recipient]
+    /// @param from Where to take liquidity for swap
+    /// @param tokenIn Input token
+    /// @param amountIn Amount of tokenIn to take for swap
+    function swapKatanaV3(
+        uint256 stream,
+        address from,
+        address tokenIn,
+        uint256 amountIn
+    ) private {
+        address pool = stream.readAddress();
+        bool direction = stream.readUint8() == DIRECTION_TOKEN0_TO_TOKEN1;
+        address recipient = stream.readAddress();
+
+        if (
+            pool == address(0) ||
+            pool == IMPOSSIBLE_POOL_ADDRESS ||
+            recipient == address(0)
+        ) revert InvalidCallData();
+
+        // get router address from pool governance
+        address governance = IKatanaV3Pool(pool).governance();
+        address router = IKatanaV3Governance(governance).getRouter();
+
+        // get pool info for constructing the path
+        address token0 = IKatanaV3Pool(pool).token0();
+        address token1 = IKatanaV3Pool(pool).token1();
+        uint24 fee = IKatanaV3Pool(pool).fee();
+
+        // determine tokenOut based on swap direction
+        address tokenOut = direction ? token1 : token0;
+
+        // transfer tokens to the router
+        if (from == msg.sender) {
+            IERC20(tokenIn).safeTransferFrom(msg.sender, router, amountIn);
```

Comment:
I would vouch to use LibAsset.transferFromERC20() here since it ensures that funds cannot be sent to address(0) (this check is missing here which could potentially mean lost funds

Reply:
Hmm...but you are still using the IERC20.safeTransferFrom here.
Did you maybe forget to update this function?

Reply:
tes, updated